RadarSDK/RadarNotificationHelper.m

```diff
@@ -108,12 +108,37 @@ - (void)swizzled_userNotificationCenter:(UNUserNotificationCenter *)center
         didReceiveNotificationResponse:(UNNotificationResponse *)response
                  withCompletionHandler:(void (^)(void))completionHandler {
 
-    [RadarNotificationHelper logConversionWithNotificationResponse:response];
+    RadarInitializeOptions *options = [RadarSettings initializeOptions];
+    if (options.autoHandleNotificationDeepLinks) {
+        [RadarNotificationHelper extractURLFromNotification:response.notification];
+    }
+    if (options.autoLogNotificationConversions) {
+        [RadarNotificationHelper logConversionWithNotificationResponse:response];
+    }
 
     // Call the original method (which is now swizzled)
     [self swizzled_userNotificationCenter:center didReceiveNotificationResponse:response withCompletionHandler:completionHandler];
 }
 
++ (void)extractURLFromNotification:(UNNotification *)notification {
+
+    if ([notification.request.identifier hasPrefix:@"radar_"]) {
+        NSString *urlString = notification.request.content.userInfo[@"url"];
+        if (urlString) {
+            NSURL *url = [NSURL URLWithString:urlString];
+            if (url) {
+                if([[RadarDelegateHolder sharedInstance] didHandleURL:url]) {
+                    return;
+                }
+                UIApplication *application = [UIApplication sharedApplication];
+                if ([application canOpenURL:url]) {
+                   [application openURL:url options:@{} completionHandler:nil];
```

Comment:
this is deprecated: https://developer.apple.com/documentation/uikit/uiapplication/1622961-openurl

looks like this is how we'd want to do this: https://developer.apple.com/documentation/uikit/uiapplication/1648685-open

Reply:
I think I'm using the right objective C version of it: https://developer.apple.com/documentation/uikit/uiapplication/1648685-openurl?language=objc